terraform/variables.go

```diff
@@ -5,21 +5,51 @@ import (
 	"github.com/zclconf/go-cty/cty"
 )
 
+type VariableMap map[string]cty.Value
+
 func extractFromFile(file *hcl.File) (VariableMap, hcl.Diagnostics) {
-	// TODO: remove temporary dummy variables
-	// implement "var" when we do input variables and default values
+	// TODO: remove temporary dummy locals
 	// implement "local" when we do local values
+
+	varMap, hclDiags := extractFromTfFile(file)
+	if hclDiags.HasErrors() {
+		return VariableMap{}, hclDiags
+	}
+
 	return VariableMap{
-		"var": cty.ObjectVal(VariableMap{
-			"dummy": cty.StringVal("dummy_value"),
-		}),
+		"var": cty.ObjectVal(varMap),
 		"local": cty.ObjectVal(VariableMap{
-			"dummy": cty.StringVal("dummy_local"),
+			"dummy": cty.StringVal("dummy"),
 		}),
-	}, nil
+	}, hclDiags
 }
 
-type VariableMap map[string]cty.Value
+func extractFromTfFile(file *hcl.File) (VariableMap, hcl.Diagnostics) {
```

Comment:
nit: Would it be nicer to name this `extractVariableBlocks` or something like this?

Reply:
I think that `extractVariableBlocks` sounds more abstract but I personally think that `extractFromTfFile` makes it more explicit what this function does and what it's being used for.